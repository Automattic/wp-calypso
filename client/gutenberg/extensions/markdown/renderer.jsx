/** @format */

/**
 * External dependencies
 */
import MarkdownIt from 'markdown-it';
import { __ } from '@wordpress/i18n';
import { RawHTML } from '@wordpress/element';

/**
 * Module variables
 */
const markdownConverter = new MarkdownIt();
const handleLinkClick = event => {
	if ( event.target.nodeName === 'A' ) {
		const hasConfirmed = window.confirm(
			__( 'Are you sure you wish to leave this page?', 'jetpack' )
		);

		if ( ! hasConfirmed ) {
			event.preventDefault();
		}
	}
};

export default ( { className, source = '' } ) => (
	<RawHTML className={ className } onClick={ handleLinkClick }>
		{ source.length ? markdownConverter.render( source ) : '' }
	</RawHTML>
);
