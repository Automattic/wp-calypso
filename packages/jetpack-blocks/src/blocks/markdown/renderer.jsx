/**
 * External dependencies
 */
import MarkdownIt from 'markdown-it';
import { RawHTML } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { __ } from '../../utils/i18n';

/**
 * Module variables
 */
const markdownConverter = new MarkdownIt();
const handleLinkClick = event => {
	if ( event.target.nodeName === 'A' ) {
		const hasConfirmed = window.confirm( __( 'Are you sure you wish to leave this page?' ) );

		if ( ! hasConfirmed ) {
			event.preventDefault();
		}
	}
};

export default ( { className, source = '' } ) => (
	<RawHTML className={ className } onClick={ handleLinkClick }>
		{ source.length ? markdownConverter.render( source ) : '' }
	</RawHTML>
);
