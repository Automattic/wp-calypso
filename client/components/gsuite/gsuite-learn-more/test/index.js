/** @format */

/**
 * External dependencies
 */
import React from 'react';

import renderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import GSuiteLearnMore from '../';

describe( 'GSuiteLearnMore', () => {
	test( 'it renders GSuiteLearnMore with no props', () => {
		const tree = renderer.create( <GSuiteLearnMore /> ).toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
