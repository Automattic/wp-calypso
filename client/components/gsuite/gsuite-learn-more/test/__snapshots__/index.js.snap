// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GSuiteLearnMore it renders GSuiteLearnMore with no props 1`] = `
<div
  className="gsuite-learn-more"
>
  <p>
    <strong>
      No setup or software required.
    </strong>
     
    <a
      className="gsuite-learn-more__link"
      href="https://wordpress.com/support/add-email/adding-g-suite-to-your-site/"
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn more about integrating G Suite with your site.
    </a>
  </p>
</div>
`;
